.gitlab/config/generate_pipelines.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+# Distributed under the OSI-approved Apache License, Version 2.0.  See
+# accompanying file Copyright.txt for details.
+#
+# generate_pipeline.py
+#
+#  Created: May 19, 2023
+#   Author: Vicente Adolfo Bolea Sanchez <vicente.bolea@kitware.com>
+
+from datetime import datetime
+import argparse
+import requests
+import time
+import re
+import urllib3
+# Remove annoying warning about insecure connection (self-signed cert).
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def is_date_after(date, days):
+    deadline_sec = int(time.time()) - (days * 86400)
+    utc_dt = datetime.strptime(date, '%Y-%m-%dT%H:%M:%SZ')
+    timestamp_sec = (utc_dt - datetime(1970, 1, 1)).total_seconds()
+    return timestamp_sec > deadline_sec
+
+
+def request_dict(url):
+    r = requests.get(url + '?per_page=100', verify=False)
```

Comment:
Thanks for adding the `per_page` parameter, that should allow us to handle paging in a later PR.  Or maybe we don't even need to worry about it with the current PR volume.